client-app/components/atoms/image-gallery/image-gallery.vue

```diff
@@ -0,0 +1,80 @@
+<template>
+  <div
+    ref="mainImageDiv"
+    class="square relative flex flex-col justify-center items-center border border-gray-100 rounded-sm"
+  >
+    <img
+      :src="activeSrc"
+      alt="product.name"
+      class="absolute top-0 w-full h-full object-cover object-center rounded-sm"
+    />
+  </div>
+  <div v-if="isMobile && images && images.length > 1" class="mt-4 flex flex-row justify-center space-x-2.5">
+    <div
+      v-for="image in images"
+      :key="image?.url || ''"
+      class="border border-yellow-500 w-4 h-4 rounded-full cursor-pointer"
+      :class="{
+        'bg-yellow-500': image?.url == activeSrc,
+      }"
+      @click="setActiveImage(image?.url || '')"
```

Comment:
Here you should pass `image?.url` and handle undefined value inside `setActiveImage` function. Please, don't overcomplicate template logic.